bot/exts/utils/clean.py

```diff
@@ -121,54 +229,68 @@ def predicate_regex(message: Message) -> bool:
             # Invocation message has already been deleted
             log.info("Tried to delete invocation message, but it was already deleted.")
 
-        messages = []
-        message_ids = []
         self.cleaning = True
 
-        # Find the IDs of the messages to delete. IDs are needed in order to ignore mod log events.
-        for channel in channels:
-            async for message in channel.history(limit=amount):
+        if use_cache:
+            message_mappings, message_ids = self._get_messages_from_cache(amount=amount, predicate=predicate)
+        else:
+            message_mappings, message_ids = await self._get_messages_from_channels(
+                amount=amount,
+                channels=channels,
+                predicate=predicate,
+                until_message=until_message
+            )
+
+        if not self.cleaning:
+            # Means that the cleaning was canceled
+            return
+
+        # Now let's delete the actual messages with purge.
+        self.mod_log.ignore(Event.message_delete, *message_ids)
+
+        # Creates ID like int object that would represent an object that is exactly 14 days old
+        minimum_time = int((time.time() - 14 * 24 * 60 * 60) * 1000.0 - 1420070400000) << 22
```

Comment:
I would change the name of the variable to better document the meaning of the comparison a few lines below. Something like `two_week_old_snowflake`.

Reply:
Also, is there a reason to not check the age of the message instead?

Reply:
> Also, is there a reason to not check the age of the message instead?
This part of the code was inspired by how `purge` works internally.
Comparison on message age could possibly be less accurate which in turn would resort in problems with message deletion if said messages are very close to the 14d mark.

Reply:
Might be a little less readable this way, although if there's a known limitation I suppose it's fine. In that case though this kind of decision should be added to the comment above the line.